orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -64,6 +75,36 @@ public T start(T execution) throws Exception {
     return execution;
   }
 
+  protected T handleStartupFailure(T execution, Throwable failure) {
+    final String canceledBy = "system";
+    final String reason = "Failed on startup: " + failure.getMessage();
+    final ExecutionStatus status = ExecutionStatus.TERMINAL;
+    executionRepository.updateStatus(execution.getId(), status);
+    Function<Execution, Execution> reloader;
+    String executionType;
+    if (execution instanceof Pipeline) {
+      executionType = "pipeline";
+      reloader = (e) -> executionRepository.retrievePipeline(e.getId());
+    } else if (execution instanceof Orchestration) {
+      executionType = "orchestration";
+      reloader = (e) -> executionRepository.retrieveOrchestration(e.getId());
+    } else {
+      log.error("Unknown execution type: " + execution.getClass().getSimpleName());
+      executionType = "unknown";
+      reloader = (e) -> {
+        e.setCancellationReason(reason);
+        e.setCanceled(true);
+        e.setCanceledBy(canceledBy);
+        e.setStatus(status);
+        return e;
+      };
```

Comment:
This block is trying to reload the execution after the cancel happens on it (since the repository does an update to the backing store by Id and we have a full object), and because there are dedicated getOrchestration and getPipeline methods (which are used if we have a known type) the else case there is really just there to handle the case of us adding a 3rd type of Execution (we would never do that right).